pkg/controller/egress_router/egress_router_controller.go

```diff
@@ -0,0 +1,212 @@
+		boolTrue := bool(true)
+		EgressRouterOwnerReferences := []v1.OwnerReference{
+			{
+				APIVersion: "network.operator.openshift.io/v1",
+				Kind:       "EgressRouter",
+				Name:       obj.Name,
+				UID:        obj.UID,
+				Controller: &boolTrue,
+			},
+		}
+		objs, err := ensureEgressRouter(manifestDir, request.Namespace, obj)
+		for _, obj := range objs {
+			log.Printf("Assigning owner references")
+			obj.SetOwnerReferences(EgressRouterOwnerReferences)
+			log.Printf("Applying manifest")
+			if err := apply.ApplyObject(context.TODO(), r.client, obj); err != nil {
+				log.Printf("could not apply operator config: %v", err)
+				r.status.SetDegraded(statusmanager.EgressRouterConfig, "UpdateEgressRouterConfig",
+					fmt.Sprintf("Error while reconciling EgressRouter: %v", err))
+			}
+		}
+
+		if err != nil {
+			log.Println(err)
+			r.egressrouterErrs[request.NamespacedName] =
+				errors.Wrapf(err, "could not reconcile Egress Router %s", request.NamespacedName)
+			r.setStatus()
+			return reconcile.Result{}, err
+		}
+		r.egressrouters[request.NamespacedName] = existing
+	}
+
+	if err != nil {
+		log.Println(err)
+		r.egressrouterErrs[request.NamespacedName] =
+			errors.Wrapf(err, "could not reconcile Egress Router %s", request.NamespacedName)
+		r.setStatus()
+		return reconcile.Result{}, err
+	}
+
+	log.Println("successful reconciliation")
+	delete(r.egressrouterErrs, request.NamespacedName)
+	r.setStatus()
+	return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
+}
+
+// setStatus summarizes the status of all Egress Router objects and updates the statusmanager
+// as appropriate.
+func (r *EgressRouterReconciler) setStatus() {
+	if len(r.egressrouterErrs) == 0 {
+		r.status.SetNotDegraded(statusmanager.EgressRouterConfig)
+	} else {
+		msgs := []string{}
+		for _, e := range r.egressrouterErrs {
+			msgs = append(msgs, e.Error())
+		}
+		r.status.SetDegraded(statusmanager.EgressRouterConfig, "EgressRouterError", strings.Join(msgs, ", "))
+	}
+}
+
+func isItValidCidr(cidr string) bool {
+	_, _, err := net.ParseCIDR(cidr)
+	if err != nil {
+		log.Println(err)
+		return false
+	}
+	return true
+}
+
+func isItValidIPAddress(ip string) bool {
+	return net.ParseIP(ip) != nil
+}
+
+func ensureEgressRouter(manifestDir string, namespace string, router *netopv1.EgressRouter) ([]*uns.Unstructured, error) {
+	out := []*uns.Unstructured{}
+	data := render.MakeRenderData()
+	data.Data["ReleaseVersion"] = os.Getenv("RELEASE_VERSION")
+	data.Data["EgressRouterNamespace"] = namespace
+	if isItValidCidr(string(router.Spec.Addresses[0].IP)) {
+		data.Data["Addresses"] = router.Spec.Addresses[0].IP
+	}
+	if isItValidIPAddress(router.Spec.Addresses[0].Gateway) {
+		data.Data["Gateway"] = router.Spec.Addresses[0].Gateway
+	}
+	data.Data["AllowedDestinations"] = router.Spec.AllowedDestinations
+	data.Data["mode"] = router.Spec.Mode
+	data.Data["network_interfaces"] = router.Spec.NetworkInterface
+	data.Data["EgressRouterPodImage"] = "centos/tools"
```

Comment:
I'd double-check if this is valid for release.
When I did the pod network connectivity check I was flagged for not using an openshift org image for doing curl tests, had to switch to openshift/hello-openshift.

Reply:
Thanks for checking that, just in case I'll make it match with the one in [1] (registry.redhat.com/openshift3/ose-pod)

[1] https://docs.openshift.com/container-platform/4.7/networking/ovn_kubernetes_network_provider/deploying-egress-router-ovn-redirection.html